packages/frontend/src/components/screens/WelcomeScreen/index.tsx

```diff
@@ -35,20 +42,52 @@ export default function WelcomeScreen({ selectedAccountId, ...props }: Props) {
     checkAccounts()
   }, [])
 
+  /**
+   * this function is called when the back button is clicked
+   * but also if the dialog is closed by pressing esc multiple
+   * times, which will force Chrome to close the dialog
+   * see https://issues.chromium.org/issues/346597066
+   *
+   * it will cancel the account creation process and call
+   * onExitWelcomeScreen
+   */
+  const onClickBackButton = async () => {
+    try {
+      const acInfo = await BackendRemote.rpc.getAccountInfo(selectedAccountId)
+      if (acInfo.kind === 'Unconfigured') {
+        await props.onUnSelectAccount()
+        await EffectfulBackendActions.removeAccount(selectedAccountId)
+      }
+      props.onExitWelcomeScreen()
+    } catch (error) {
+      if (error instanceof Error) {
+        openDialog(AlertDialog, {
+          message: error?.message,
+          cb: () => {},
+        })
+      } else {
+        log.error('unexpected error type', error)
+        throw error
+      }
+    }
+  }
+
   return (
     <ImageBackdrop variant='welcome'>
       <Dialog
         fixed
         width={400}
         canEscapeKeyClose={false}
         canOutsideClickClose={false}
+        onClose={onClickBackButton}
```

Comment:
Hmm but doesn't this get invoked also when the dialog gets closed normally, e.g. with "Submit"?

Reply:
No it's not invoked. The dialog is replaced with openDialog(ConfigureProgressDialog) and that does not trigger the onClose of the initial Dialog. (which is or is not expected behaviour?)

It seems while the double escape triggers the onClose event, just removing the old dialog from DOM does not trigger it...